File: 27-remove-element.scala
object Solution {
  def removeElement(nums: Array[Int], `val`: Int): Int = {
    var i = 0
    for (j <- nums.indices) {
      if (nums(j) != `val`) {
        nums(i) = nums(j)
        i += 1
      }
    }
    i
  }
}

File: 27-remove-element.ts
/* https://leetcode.com/problems/remove-element/ */
function removeElement(nums: number[], val: number): number {
    for (var i = 0; i < nums.length; i++) {
        if (nums[i] === val) {
            nums.splice(i, 1);
            i--;
        }
    }
    return nums.length;
};